grakn-graph/src/main/java/ai/grakn/graph/internal/ConceptCache.java

```diff
@@ -39,8 +39,14 @@
  *
  */
 class ConceptCache<V> {
+    //If no cache can produce the data then the database is read
     private final Supplier<V> databaseReader;
-    private Optional<V> cachedValue = Optional.empty();
+
+    //Transaction bound. If this is not set it does not yet exist in the graph
+    private ThreadLocal<V> cachedValue = new ThreadLocal<>();
```

Comment:
Not sure what this comment means. Isn't it possible for something to exist in the graph, but not be set in the cache?